pkg/provider/cloud/baremetal/provider.go

```diff
@@ -32,11 +35,36 @@ func NewCloudProvider() provider.CloudProvider {
 
 var _ provider.CloudProvider = &baremetal{}
 
-func (b *baremetal) DefaultCloudSpec(_ context.Context, _ *kubermaticv1.ClusterSpec) error {
+func (b *baremetal) DefaultCloudSpec(_ context.Context, spec *kubermaticv1.ClusterSpec) error {
+	if spec.Cloud.Baremetal == nil {
+		return errors.New("baremetal cloud provider spec is empty")
+	}
+
+	if spec.Cloud.Baremetal != nil && spec.Cloud.Baremetal.Tinkerbell == nil {
+		return errors.New("tinkerbell spec is empty! we must have a provisioner")
```

Comment:
please rephrase, no "we must have!", more like "tinkerbell spec is required" or something along those lines.

Also the != nil comparison here is redundant due to the first if statement.

Reply:
Also, why is the validation logic in the defaulting function? ^^

Reply:
After checking, kubevirt and aws providers. I found that there is a check on the provider object if it's null in DefaultCloudSpec()

Reply:
Error message is still unchanged.